### PR TITLE
CI: use scripts for docker build

### PR DESCRIPTION
**Pull request by jnewb1**
this will allow you to test the building process locally, and also reduces copy and paste

### CODE REVIEW

**Comment by jnewb1** [comment]
PR Comments will fail until we merge to master, because it's trying to load the build script that isn't there

File: .github/scripts/docker_build.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+if   [ $1 = "base" ]; then
+    export DOCKER_IMAGE=openpilot-base
```
Comment (adeebshihadeh) [inline .github/scripts/docker_build.sh:4]: wrong indentation

File: .github/scripts/docker_build.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
```
Comment (adeebshihadeh) [inline .github/scripts/docker_build.sh:1]:
```suggestion
#!/bin/bash
set -e
```

Comment (adeebshihadeh) [inline .github/scripts/docker_build.sh:1]: can you put this file in selfdrive/test?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
Ran this and it didn't work:
```
batman:scripts$ ./docker_build.sh base
[+] Building 0.1s (2/2) FINISHED
 => [internal] load build definition from Dockerfile.openpilot_base                                               0.0s
 => => transferring dockerfile: 2B                                                                                0.0s
 => [internal] load .dockerignore                                                                                 0.0s
 => => transferring context: 2B                                                                                   0.0s
ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount3681130445/Dockerfile.openpilot_base: no such file or directory
batman:scripts$
```

**Comment by jnewb1** [comment]
> Ran this and it didn't work:
> 
> ```
> batman:scripts$ ./docker_build.sh base
> [+] Building 0.1s (2/2) FINISHED
>  => [internal] load build definition from Dockerfile.openpilot_base                                               0.0s
>  => => transferring dockerfile: 2B                                                                                0.0s
>  => [internal] load .dockerignore                                                                                 0.0s
>  => => transferring context: 2B                                                                                   0.0s
> ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount3681130445/Dockerfile.openpilot_base: no such file or directory
> batman:scripts$
> ```

have to run from root directory

File: .github/scripts/docker_build.sh (hunk 2 above)
Comment (jnewb1) [reply .github/scripts/docker_build.sh]: done

Comment (jnewb1) [reply .github/scripts/docker_build.sh]: done

File: .github/scripts/docker_build.sh (hunk 1 above)
Comment (jnewb1) [reply .github/scripts/docker_build.sh]: done

**Comment by jnewb1** [comment]
made it work from any directory

File: .github/workflows/prebuilt.yaml [hunk 3]
```diff
@@ -5,20 +5,16 @@ on:
   workflow_dispatch:
 
 env:
-  BASE_IMAGE: openpilot-base
-  DOCKER_REGISTRY: ghcr.io/commaai
-
   DOCKER_LOGIN: docker login ghcr.io -u ${{ github.actor }} -p ${{ secrets.GITHUB_TOKEN }}
-  BUILD: |
-      DOCKER_BUILDKIT=1 docker build --pull --cache-to type=inline --cache-from $DOCKER_REGISTRY/$BASE_IMAGE:latest -t $DOCKER_REGISTRY/$BASE_IMAGE:latest -t $BASE_IMAGE:latest -f Dockerfile.openpilot_base .
+  BUILD: selfdrive/test/docker_build.sh prebuilt
```
Comment (adeebshihadeh) [inline .github/workflows/prebuilt.yaml:9]: no need for the environment variable anymore, just adds another level of indirection

File: .github/workflows/selfdrive_tests.yaml [hunk 4]
```diff
@@ -12,20 +12,17 @@ concurrency:
...
   DOCKER_LOGIN: docker login ghcr.io -u ${{ github.actor }} -p ${{ secrets.GITHUB_TOKEN }}
-  BUILD: |
-      DOCKER_BUILDKIT=1 docker build --pull --cache-to type=inline --cache-from $DOCKER_REGISTRY/$BASE_IMAGE:latest -t $DOCKER_REGISTRY/$BASE_IMAGE:latest -t $BASE_IMAGE:latest -f Dockerfile.openpilot_base .
+  BUILD: selfdrive/test/docker_build.sh base
 
   RUN: docker run --shm-size 1G -v $PWD:/tmp/openpilot -w /tmp/openpilot -e PYTHONWARNINGS=error -e FILEREADER_CACHE=1 -e PYTHONPATH=/tmp/openpilot -e NUM_JOBS -e JOB_ID -e GITHUB_ACTION -e GITHUB_REF -e GITHUB_HEAD_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -v ~/scons_cache:/tmp/scons_cache -v ~/comma_download_cache:/tmp/comma_download_cache -v ~/openpilot_cache:/tmp/openpilot_cache $BASE_IMAGE /bin/sh -c
 
-  BUILD_CL: |
-      DOCKER_BUILDKIT=1 docker build --cache-to type=inline --cache-from $DOCKER_REGISTRY/$CL_BASE_IMAGE:latest -t $DOCKER_REGISTRY/$CL_BASE_IMAGE:latest -t $CL_BASE_IMAGE:latest -f Dockerfile.openpilot_base_cl .
+  BUILD_CL: selfdrive/test/docker_build.sh cl
```
Comment (adeebshihadeh) [inline .github/workflows/selfdrive_tests.yaml:24]: same here

File: .github/workflows/tools_tests.yaml [hunk 5]
```diff
@@ -13,16 +13,14 @@ concurrency:
...
+  BUILD: selfdrive/test/docker_build.sh base
 
   RUN: docker run --shm-size 1G -v $GITHUB_WORKSPACE:/tmp/openpilot -w /tmp/openpilot -e FILEREADER_CACHE=1 -e PYTHONPATH=/tmp/openpilot -e NUM_JOBS -e JOB_ID -e GITHUB_ACTION -e GITHUB_REF -e GITHUB_HEAD_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -v ~/scons_cache:/tmp/scons_cache -v ~/comma_download_cache:/tmp/comma_download_cache -v ~/openpilot_cache:/tmp/openpilot_cache $BASE_IMAGE /bin/sh -c
 
-  BUILD_CL: |
-      DOCKER_BUILDKIT=1 docker build --pull --cache-to type=inline --cache-from $DOCKER_REGISTRY/$CL_BASE_IMAGE:latest -t $DOCKER_REGISTRY/$CL_BASE_IMAGE:latest -t $CL_BASE_IMAGE:latest -f Dockerfile.openpilot_base_cl .
+  BUILD_CL: selfdrive/test/docker_build.sh cl
+  
   RUN_CL: docker run --shm-size 1G -v $GITHUB_WORKSPACE:/tmp/openpilot -w /tmp/openpilot -e PYTHONPATH=/tmp/openpilot -e NUM_JOBS -e JOB_ID -e GITHUB_ACTION -e GITHUB_REF -e GITHUB_HEAD_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -v ~/scons_cache:/tmp/scons_cache -v ~/comma_download_cache:/tmp/comma_download_cache -v ~/openpilot_cache:/tmp/openpilot_cache $CL_BASE_IMAGE /bin/sh -c
```
Comment (adeebshihadeh) [inline .github/workflows/tools_tests.yaml:24]: we should also do this for the docker runs

File: selfdrive/test/docker_build.sh [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+elif [ $1 = "cl" ]; then
+  export DOCKER_IMAGE=openpilot-base-cl
+  export DOCKER_FILE=Dockerfile.openpilot_base_cl
+else
+  echo "Invalid docker build image $1"
+  exit 1
+fi
+
+export DOCKER_REGISTRY=ghcr.io/commaai
+export COMMIT_SHA=$(git rev-parse HEAD);
```
Comment (adeebshihadeh) [inline selfdrive/test/docker_build.sh:25]: what's the semicolon for?

File: selfdrive/test/docker_build.sh [hunk 7]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -e
+
+if   [ $1 = "base" ]; then
```
Comment (adeebshihadeh) [inline selfdrive/test/docker_build.sh:4]: too many spaces

File: .github/workflows/prebuilt.yaml (hunk 3 above)
Comment (jnewb1) [reply .github/workflows/prebuilt.yaml:9]: that will require refactoring the setup script to take a parameter, which should probably be a separate PR

File: .github/workflows/tools_tests.yaml (hunk 5 above)
Comment (jnewb1) [reply .github/workflows/tools_tests.yaml:24]: separate PR?

File: selfdrive/test/docker_build.sh (hunk 7 above)
Comment (jnewb1) [reply selfdrive/test/docker_build.sh]: done

File: selfdrive/test/docker_build.sh (hunk 6 above)
Comment (jnewb1) [reply selfdrive/test/docker_build.sh]: done

File: .github/workflows/tools_tests.yaml (hunk 5 above)
Comment (adeebshihadeh) [reply .github/workflows/tools_tests.yaml:24]: sure

**Review by adeebshihadeh — APPROVED** [review]

**Pull request merged**